yarn-project/acir-simulator/src/avm/avm_memory_types.ts

```diff
@@ -220,11 +220,16 @@ export enum TypeTag {
 
 // TODO: Consider automatic conversion when getting undefined values.
 export class TaggedMemory {
-  static readonly MAX_MEMORY_SIZE = 1n << 32n;
+  // FIXME: memory should be 2^32, but TS doesn't allow for arrays that big.
+  static readonly MAX_MEMORY_SIZE = Number(1n << 31n); // 1n << 32n
   private _mem: MemoryValue[];
 
   constructor() {
     this._mem = [];
+    // The following two lines basically initialize a sparse array of length MAX_MEMORY_SIZE.
+    // All entries will be undefined afterwards.
+    this._mem[TaggedMemory.MAX_MEMORY_SIZE] = new Field(0);
```

Comment:
Would this._mem = new Array(size) or something not work?

Reply:
It might.... Goal of this is to basically leave the entire array uninitialized and sparse.... Maybe I over thought this 😅. I'll try a simple array constructor

Reply:
We don't want to actually store the entire uninitialized mem in memory.... But actually now that I think of it, maybe new Array will also initialize it in a sparse manner

Reply:
When I saw the original code using an array (well, []) as memory I was also surprised, since assigning to `TaggedMemory.MAX_MEMORY_SIZE` (or any big number) might just resize to that size. So I looked for the spec (well, some websites) and it looks like JS manages the size. If the array is small, it's a chunk of mem; if it's big, it's a hashmap.

Why are these two lines needed? What I mention above should happen automatically as it grows.

Reply:
2^32 is ts max size so it probably overflows it into a 0

Reply:
fwiw, [] was just to get this off the ground, swapping it to a map or whatever is fine in my eyes